tests/fancy_pointer/sparse_hash_set_tests.cpp

```diff
@@ -0,0 +1,184 @@
+     *  So basically Set \subset l and |Set| == |l|.
+     *  Needs 'set.contains(.)' to work correctly.
+     */
+    template <typename Set>
+    bool is_equal(Set const& set, std::initializer_list<typename Set::value_type> l) {
+        return std::all_of(l.begin(), l.end(), [&set](typename Set::value_type i){return set.contains(i);})
+            and set.size() == l.size();
+    }
+}
+
+template<typename T>
+void construction() {
+    auto set = details::default_construct_set<T>();
+}
+
+template <typename T>
+void insert(std::initializer_list<typename T::value_type> l) {
+    auto set = details::default_construct_set<T>();
+    for (auto const& i: l)  set.insert(i);
+    BOOST_TEST_REQUIRE(details::is_equal(set, l), "'insert' did not create exactly the values needed");
+}
+
+template <typename T>
+void iterator_insert(std::initializer_list<typename T::value_type> l) {
+   auto set = details::default_construct_set<T>();
+   set.insert(l.begin(), l.end());
+   BOOST_TEST_REQUIRE(details::is_equal(set, l), "'insert' with iterators did not create exactly the values needed");
+}
+
+template <typename T>
+void iterator_access(typename T::value_type single_value) {
+    auto set = details::default_construct_set<T>();
+    set.insert(single_value);
+    BOOST_TEST_REQUIRE(*(set.begin()) == single_value, "iterator cannot access single value");
+}
+
+template <typename T>
+void iterator_access_multi(std::initializer_list<typename T::value_type> l) {
+    auto set = details::default_construct_set<T>();
+    set.insert(l.begin(), l.end());
+    std::vector<typename T::value_type> l_sorted = l;
+    std::vector<typename T::value_type> set_sorted(set.begin(), set.end());
+    std::sort(l_sorted.begin(), l_sorted.end());
+    std::sort(set_sorted.begin(), set_sorted.end());
+    BOOST_TEST_REQUIRE(std::equal(l_sorted.begin(), l_sorted.end(),
+                                  set_sorted.begin()),
+                       "iterating over the set didn't work");
+}
+
+
+template <typename T>
+void const_iterator_access_multi(std::initializer_list<typename T::value_type> l) {
+    auto set = details::default_construct_set<T>();
+    set.insert(l.begin(), l.end());
+    std::vector<typename T::value_type> l_sorted = l;
+    std::vector<typename T::value_type> set_sorted(set.cbegin(), set.cend());
+    std::sort(l_sorted.begin(), l_sorted.end());
+    std::sort(set_sorted.begin(), set_sorted.end());
+    BOOST_TEST_REQUIRE(std::equal(l_sorted.begin(), l_sorted.end(),
+                                  set_sorted.begin()),
+                       "const iterating over the set didn't work");
+}
+
+template <typename T>
+void erase(std::initializer_list<typename T::value_type> l, typename T::value_type extra_value) {
+    auto set = details::default_construct_set<T>();
+    set.insert(extra_value);
+    set.insert(l.begin(), l.end());
+    //find is not tested yet
```

Comment:
todo: test find

Reply:
The comment wasn't updated yet. find is already tested, however the code forces the  non-const version of the iterator so find cannot be used.